test/perf/threads/lbm3d/circshift.jl

```diff
@@ -11,26 +10,38 @@ function circshift3d1!(a,index,vec)
     ny = size(a,2)
     nz = size(a,3)
     for i = 1 : length(vec)
-        @match vec[i] begin
-            1  => shift1!(a,nx,ny,nz,index,i)
-           -1  => shiftm1!(a,nx,ny,nz,index,i)
+        vi = vec[i]
+        if vi == 1
+            shift1!(a,nx,ny,nz,index,i)
+        elseif vi == -1
+            shiftm1!(a,nx,ny,nz,index,i)
+        else
+            throw(ArgumentError)
```

Comment:
The comment that describes this function is misleading - `vec` can contain `0` values (e.g. no shift). See [here](https://github.com/JuliaLang/julia/blob/hayd-patch-1/test/perf/threads/lbm3d/lbm3d.jl#L27).